tbx/static_src/javascript/components/table-hint.js

```diff
@@ -0,0 +1,35 @@
+// Button to scroll a table at mobile.
+// Used in conjunction with semi-opaque styling that is removed,
+// along with the button, once the user has scrolled
+// This identical to typed-table-hint.js, but due to limitations of the kit features we need to duplicate it.
+// If you have both table types in your repo, you can rationalise this to a single file.
+class TableHint {
+    static selector() {
+        return '[data-table-hint]';
+    }
+
+    constructor(node) {
+        this.node = node;
+        this.button = node.querySelector('[data-table-hint-button]');
+        this.bindEvents();
+    }
+
+    bindEvents() {
+        // Once the user scrolls, remove the button and hint and don't reshow them
+        this.node.addEventListener('scroll', () => {
+            if (this.node.scrollLeft > 0) {
+                this.node.classList.add('is-scrolling');
+            }
+        });
+
+        this.button.addEventListener('click', () => {
+            this.node.scroll({
+                top: 0,
+                left: 500,
+                behavior: 'smooth',
```

Comment:
Q: Does this need an alternative if the user has a reduced motion preference?

Reply:
good spot - will change that, and the transition in the css